components/MachineList/index.js

```diff
@@ -5,12 +5,14 @@ import { useRouter } from "next/router";
 
 export default function MachineList() {
   const router = useRouter();
-  const { data, error } = useSWR("/api/machines");
+  const { data, isLoading, error } = useSWR("/api/machines");
+
+  if (isLoading) return <div>Loading...</div>;
```

Comment:
Personal preference: use explicit if blocks instead of one liners:

```js
if (isLoading) {
  return <div>Loading...</div>;
}
```

Why? In case you need to log anything in the if, you have place to do so ^^

No big issue though, you can let it go.